server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -649,6 +649,7 @@ public void checkLeakShuffleData() {
   public void removeResources(String appId) {
     LOG.info("Start remove resource for appId[" + appId + "]");
     final long start = System.currentTimeMillis();
+    String user = getUserByAppId(appId);
```

Comment:
`user` may be null some times If app is removed.

Reply:
Only the next line code will remove shuffleTaskInfos. I move before it. 
Because after next line, getUserByAppId(appId) is always null. 
User name is not a problem for non security cluster. But user name is important for security cluster.

Reply:
> `user` may be null some times If app is removed.

@jerqi 
Only next line will remove app from shuffleTaskInfo. 
Here I move getUserByAppId before 'shuffleTaskInfos.remove(appId)', or user will be null.

Reply:
If the method can handle the null value situation, I'm ok for this change. Let @zuston take another look.